src/player.cpp

```diff
@@ -5114,6 +5141,138 @@ int player::hitall(int dam, int vary, Creature *source)
     return damage_taken;
 }
 
+float player::fall_damage_mod() const
+{
+    float ret = 1.0f;
+
+    // Ability to land properly is 2x as important as dexterity itself
+    float dex_dodge = dex_cur / 2 + get_skill_level( "dodge" );
+    // Penalize for wearing heavy stuff
+    dex_dodge -= ( ( encumb(bp_leg_l) + encumb(bp_leg_r) ) / 20 ) + ( encumb(bp_torso) / 10 );
+    // But prevent it from increasing damage
+    dex_dodge = std::min( 0.0f, dex_dodge );
+    // 100% damage at 0, 75% at 10, 50% at 20 and so on
+    ret *= (100.0f - (dex_dodge * 4.0f)) / 100.0f;
+
+    if( has_trait("PARKOUR") ) {
+        ret *= 2.0f / 3.0f;
+    }
+
+    // TODO: Bonus for Judo, mutations. Penalty for heavy weight (including mutations)
+    return std::max( 0.0f, ret );
+}
+
+// force is maximum damage to hp before scaling
+int player::impact( const int force, const tripoint &p )
+{
+    // Falls over ~30m are fatal more often than not
+    // But that would be quite a lot considering 21 z-levels in game
+    // so let's assume 1 z-level is comparable to 30 force
+
+    if( force <= 0 ) {
+        return force;
+    }
+
+    // Damage modifier (post armor)
+    float mod = 1.0f;
+    int effective_force = force;
+    int cut = 0;
+    // Percentage arpen - armor won't help much here
+    // TODO: Make cushioned items like bike helmets help more
+    float armor_eff = 1.0f;
+
+    // Being slammed against things rather than landing means we can't
+    // control the impact as well
+    const bool slam = p != pos();
+    std::string target_name = "a swarm of bugs";
+    Creature *critter = g->critter_at( p );
+    int part_num = -1;
+    vehicle *veh = g->m.veh_at( p, part_num );
+    if( critter != this && critter != nullptr ) {
+        target_name = critter->disp_name();
+        // Slamming into creatures and NPCs
+        // TODO: Handle spikes/horns and hard materials
+        armor_eff = 0.5f; // 2x as much as with the ground
+        // TODO: Modify based on something?
+        mod = 1.0f;
+        effective_force = force;
+    } else if( veh != nullptr ) {
+        // Slamming into vehicles
+        // TODO: Integrate it with vehicle collision function somehow
+        target_name = veh->name;
+        if( veh->part_with_feature( part_num, "SHARP" ) ) {
```

Comment:
I used that function before, just forgot to think here.
But yes, something like `index_of_feature` would probably be more informative.